### PR TITLE
api,server,engine/schema: admin listvm api clusterid

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Add clusterid parameter in listVirtualMachines API for admin

Import-Export Instances UI relies on cluster filtering for managed instances. Without clusterid param support in `listVirtualMachines` API for admins it lists all VMs in the zone.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

**Before:**
Screenshot from 2021-11-03 00-36-22

**After:**
Screenshot from 2021-11-03 00-45-50

Using cmk after changes:
```
(localcloud) 🐱 > list clusters filter=id,name,hypervisortype
{
  "cluster": [
    {
      "hypervisortype": "VMware",
      "id": "209e64fb-481f-4864-9a24-2c0a63352eee",
      "name": "p1-c1"
    },
    {
      "hypervisortype": "KVM",
      "id": "681e9c6f-70db-4725-a6eb-d7979ddf4acc",
      "name": "p1-c2"
    }
  ],
  "count": 2
}
(localcloud) 🐱 > list virtualmachines filter=id,name,hypervisor,hostid,hostname
{
  "count": 2,
  "virtualmachine": [
    {
      "hostid": "eb31f76b-bae8-404b-a0a1-3287cf827c6b",
      "hostname": "10.0.34.240",
      "hypervisor": "VMware",
      "id": "3a6f7f06-2b2c-4da7-95b2-046d1181284a",
      "name": "vmw-vm2"
    },
    {
      "hostid": "96b347f3-abcf-427d-bb7f-a1190f94d96e",
      "hostname": "ref-trl-2130-v-M7-abhishek-kumar-kvm2",
      "hypervisor": "KVM",
      "id": "a360b8d3-bdf1-4067-842d-bee12ca17f9e",
      "name": "kvm-vm1"
    }
  ]
}
(localcloud) 🐱 > list virtualmachines filter=id,name,hypervisor,hostid,hostname clusterid=209e64fb-481f-4864-9a24-2c0a63352eee 
{
  "count": 1,
  "virtualmachine": [
    {
      "hostid": "eb31f76b-bae8-404b-a0a1-3287cf827c6b",
      "hostname": "10.0.34.240",
      "hypervisor": "VMware",
      "id": "3a6f7f06-2b2c-4da7-95b2-046d1181284a",
      "name": "vmw-vm2"
    }
  ]
}
(localcloud) 🐱 > list virtualmachines filter=id,name,hypervisor,hostid,hostname clusterid=681e9c6f-70db-4725-a6eb-d7979ddf4acc 
{
  "count": 1,
  "virtualmachine": [
    {
      "hostid": "96b347f3-abcf-427d-bb7f-a1190f94d96e",
      "hostname": "ref-trl-2130-v-M7-abhishek-kumar-kvm2",
      "hypervisor": "KVM",
      "id": "a360b8d3-bdf1-4067-842d-bee12ca17f9e",
      "name": "kvm-vm1"
    }
  ]
}
```

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan package

File: api/src/main/java/org/apache/cloudstack/api/command/admin/vm/ListVMsCmdByAdmin.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 // under the License.
 package org.apache.cloudstack.api.command.admin.vm;
 
+import org.apache.cloudstack.api.response.ClusterResponse;
```
Comment (weizhouapache) [inline api/src/main/java/org/apache/cloudstack/api/command/admin/vm/ListVMsCmdByAdmin.java:19]: move to line 28?

Comment (shwstppr) [reply api/src/main/java/org/apache/cloudstack/api/command/admin/vm/ListVMsCmdByAdmin.java]: @weizhouapache done

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan test matrix

**Comment by shwstppr** [comment]
@blueorangutan test matrix

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Review by rhtyd — APPROVED** [review]

**Review by harikrishna-patnala — APPROVED** [review]
LGTM

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
Failing listvirtualmachinesmetrics APIs fixed with last commit

```
(localcloud) 🐱 > list virtualmachinesmetrics filter=id,name,account,networkkbsread listall=true 
{
  "count": 3,
  "virtualmachine": [
    {
      "account": "ACSUser",
      "id": "5412de2d-740a-4cb1-b8bf-63646ccf4aa9",
      "name": "VM-5412de2d-740a-4cb1-b8bf-63646ccf4aa9",
      "networkkbsread": 0
    },
    {
      "account": "admin",
      "id": "3a6f7f06-2b2c-4da7-95b2-046d1181284a",
      "name": "vmw-vm2",
      "networkkbsread": 0
    },
    {
      "account": "admin",
      "id": "a360b8d3-bdf1-4067-842d-bee12ca17f9e",
      "name": "kvm-vm1",
      "networkkbsread": 0
    }
  ]
}
(localcloud) 🐱 > set profile user
Loaded server profile: user
Url:         http://localhost:8080/client/api
Username:    user
Domain:      /
API Key:     
Total APIs:  269

(user) 🐱 > sync
Discovered 269 APIs
(user) 🐱 > list virtualmachinesmetrics filter=id,name,account,networkkbsread listall=true 
{
  "count": 1,
  "virtualmachine": [
    {
      "account": "ACSUser",
      "id": "5412de2d-740a-4cb1-b8bf-63646ccf4aa9",
      "name": "VM-5412de2d-740a-4cb1-b8bf-63646ccf4aa9",
      "networkkbsread": 0
    }
  ]
}
```

**Comment by rhtyd** [comment]
@blueorangutan test matrix

**Comment by shwstppr** [comment]
@blueorangutan test matrix

**Comment by sureshanaparti** [comment]
@blueorangutan test matrix

**Review by vladimirpetrov — APPROVED** [review]
LGTM based on manual testing on environment with 3 hypervisors, VMware 6.7u3 (main), KVM CentOS 7 and XCP-NG 8.2:

(localcloud) 🐱 > list virtualmachines filter=id,name,hypervisor,hostid,hostname clusterid=20e1a60a-c5bc-4a46-9beb-89e625eff19e 
{
  "count": 1,
  "virtualmachine": [
    {
      "hostid": "a7383222-dd6b-43c8-81d6-536927cac644",
      "hostname": "ref-trl-2136-v-M7-vladimir-petrov-kvm1",
      "hypervisor": "KVM",
      "id": "0fa611c0-cfe6-4dca-922a-fabef819b679",
      "name": "VM-KVM"
    }
  ]
}
(localcloud) 🐱 > list virtualmachines filter=id,name,hypervisor,hostid,hostname clusterid=3874307d-221d-40ae-9658-6f32e8f076fe 
{
  "count": 1,
  "virtualmachine": [
    {
      "hostid": "56c7b76e-bd43-4c98-aad2-1496012c0118",
      "hostname": "ref-trl-2136-v-M7-vladimir-petrov-xs1",
      "hypervisor": "XenServer",
      "id": "c7de3f68-d531-4bec-a326-1dfc1ef84910",
      "name": "VM-XEN"
    }
  ]
}

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java [hunk 2]
```diff
@@ -30,10 +30,6 @@
 
 import javax.inject.Inject;
 
-import com.cloud.resource.icon.dao.ResourceIconDao;
-import com.cloud.server.ResourceManagerUtil;
-import com.cloud.storage.dao.VMTemplateDetailsDao;
-import com.cloud.vm.VirtualMachineManager;
```
[the remark this answers is not in the document]
Comment (shwstppr) [reply server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:36]: @GutoVeronezi not for the PR but this keeps imports in alphabetical order

**Pull request merged**